api/server/router/container/container_routes.go

```diff
@@ -101,23 +101,30 @@ func (s *containerRouter) getContainersLogs(ctx context.Context, w http.Response
 			ShowStderr: stderr,
 			Details:    httputils.BoolValue(r, "details"),
 		},
-		OutStream: w,
+		Messages: make(chan *backend.LogMessage, 1),
 	}
 
-	chStarted := make(chan struct{})
-	if err := s.backend.ContainerLogs(ctx, containerName, logsConfig, chStarted); err != nil {
-		select {
-		case <-chStarted:
-			// The client may be expecting all of the data we're sending to
-			// be multiplexed, so mux it through the Systemerr stream, which
-			// will cause the client to throw an error when demuxing
-			stdwriter := stdcopy.NewStdWriter(logsConfig.OutStream, stdcopy.Systemerr)
-			fmt.Fprintf(stdwriter, "Error running logs job: %v\n", err)
-		default:
-			return err
-		}
+	// doesn't matter what version the client is on, we're using this internally only
+	// also do we need size? i'm thinkin no we don't
+	raw, err := s.backend.ContainerInspect(containerName, false, api.DefaultVersion)
```

Comment:
We need to know if the container has a TTY so we can decide whether or not to tell httputils.WriteLogStream to multiplex logs.